src/App.tsx

```diff
@@ -1,26 +1,132 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useEffect, useMemo, useState, useRef } from 'react';
+import { addTodo, deleteTodo, getTodos, USER_ID } from './api/todos';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo } from './types/Todo';
+import { TodoFooter } from './components/TodoFooter/TodoFooter';
+import { ErrorType } from './types/ErrorType';
+// eslint-disable-next-line max-len
+import { ErrorNotification } from './components/ErrorNotification/ErrorNotification';
+import { TodoHeader } from './components/TodoHeader/TodoHeader';
+import { FilterStatus } from './types/FilterStatus';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterStatus, setFilterStatus] = useState<FilterStatus>(
+    FilterStatus.All,
+  );
+  const [errorMessage, setErrorMessage] = useState<ErrorType>(ErrorType.Empty);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const filteredTodos = useMemo(
+    () =>
+      todos.filter(todo => {
+        if (filterStatus === FilterStatus.All) {
+          return true;
+        }
+
+        return filterStatus === FilterStatus.Completed
+          ? todo.completed
+          : !todo.completed;
+      }),
+
+    [todos, filterStatus],
+  );
+
+  const activeTodosCount = todos.filter(todo => !todo.completed).length;
+  const completedTodosCount = todos.filter(todo => todo.completed).length;
+
+  const onAddTodo = async (todoTitle: string) => {
+    try {
+      setTempTodo({
+        id: 0,
+        title: todoTitle,
+        completed: false,
+        userId: USER_ID,
+      });
+      const newTodo = await addTodo({ title: todoTitle, completed: false });
+
+      setTodos(prev => [...prev, newTodo]);
+    } catch (error) {
+      setErrorMessage(ErrorType.UnableToAdd);
+      throw error;
```

Comment:
You don't need to throw an error if you don't plan to handle it in the other block. Fix in all places

Reply:
I don't get it :(
What should I do here? My solutions are not working